service/controller/resource/monitoring/prometheus/resource.go

```diff
@@ -164,100 +155,110 @@ func toPrometheus(ctx context.Context, v interface{}, config Config) (metav1.Obj
 				},
 				Key: key.AlertManagerKey(),
 			},
-			ExternalURL: externalURL.String(),
-			RoutePrefix: fmt.Sprintf("/%s", key.ClusterID(cluster)),
-			PodMetadata: &promv1.EmbeddedObjectMetadata{
-				Labels: labels,
-			},
-			// TODO will have to migrate to EnableRemoteWriteReceiver: true but need CRD changes
-			// https://github.com/prometheus-operator/prometheus-operator/pull/4633/files
-			EnableFeatures: []string{"remote-write-receiver"},
-			LogLevel:       config.LogLevel,
-			Image:          &image,
-			Web: &promv1.WebSpec{
-				PageTitle: &pageTitle,
-			},
-			Replicas: &replicas,
-			Resources: corev1.ResourceRequirements{
-				Requests: corev1.ResourceList{
-					// cpu: 100m
-					corev1.ResourceCPU: *key.PrometheusDefaultCPU(),
-					// memory: 1Gi
-					corev1.ResourceMemory: *key.PrometheusDefaultMemory(),
-				},
-				Limits: corev1.ResourceList{
-					// cpu: 150m
-					corev1.ResourceCPU: *key.PrometheusDefaultCPULimit(),
-					// memory: 1.2Gi
-					corev1.ResourceMemory: *key.PrometheusDefaultMemoryLimit(),
-				},
-			},
-			Retention:      config.RetentionDuration,
-			RetentionSize:  config.RetentionSize,
-			WALCompression: &walCompression,
-			AdditionalScrapeConfigs: &corev1.SecretKeySelector{
-				LocalObjectReference: corev1.LocalObjectReference{
-					Name: key.PrometheusAdditionalScrapeConfigsSecretName(),
+
+			CommonPrometheusFields: promv1.CommonPrometheusFields{
```

Comment:
In latest prometheus-operator, most of the common fields are shared in a CommonPrometheusFields.

This is prework so they can add PrometheusAgentCR